tdms/src/hdf5_io.cpp

```diff
@@ -60,10 +83,12 @@ std::vector<hsize_t> HDF5Base::shape_of(const std::string &dataname) const {
   int rank = dataspace.getSimpleExtentNdims();
   std::vector<hsize_t> dimensions(rank);
   dataspace.getSimpleExtentDims(dimensions.data(), nullptr);
-
-  // vector is the size in each dimension i, j(, k)
   return dimensions;
 }
 
-
-bool HDF5Base::is_ok() const { return true; }
+bool HDF5Base::is_ok() const {
+  // TODO: check for file health might be unnessicary given we've constructed
+  // the object.
+  return file_->isHdf5(filename_);
+  // return file_->isAccessible(filename_) && file_->isHdf5(filename_);
+}
```

Comment:
He's back, but I'm fairly sure he's going to disappear again in later commits.

Reply:
Not in the header though afai can see.

Reply:
Correction: it _is_ in the header. And we do actually use it.